theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
> switched the dimensions of inputs and outputs though

Oops, yes. :+1: for working on this. Regarding the choice of memory layout / shape, we might want to check other frameworks' implementations again (linked at https://github.com/Theano/Theano/issues/5621#issuecomment-282344400). If there's no clear argument for either layout, we might opt for compatibility (to make it easier to port models between frameworks).